src/RandomUuid.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef RANDOM_UUID_H
+#define RANDOM_UUID_H
+
+#include <iostream>
+
+// RFC4122: Universally Unique IDentifier (UUID)
+class RandomUuid
+{
+public:
+    RandomUuid();
+    RandomUuid(void *data, int length);
+
```

Comment:
Please prohibit public/default copying of UUIDs to catch accidental copying mistakes, especially when another class has a UUID field. Add a duplicate() method that returns a copy for those users that do need to copy a UUID.